File: AquaTrack/src/main/java/com/aquatrack/service/PropertyAdminService.java
package com.aquatrack.service;

import com.aquatrack.dto.propertyadmin.PropertyAdminActivationResponse;
import com.aquatrack.dto.propertyadmin.SetPasswordRequest;

public interface PropertyAdminService {

    // ==========================================
    // Validate Activation Token
    // ==========================================

    PropertyAdminActivationResponse validateActivationToken(
            String token
    );



    // ==========================================
    // Set Password & Activate Account
    // ==========================================

    void setPassword(
            SetPasswordRequest request
    );

}

File: Aquatrack-frontend/src/services/propertyAdminService.js
import api from "../api/axios";

// ==========================================
// Validate Activation Token
// ==========================================

export const validateActivationToken = async (token) => {

    const response = await api.get(
        `/property-admin/activate?token=${token}`
    );

    return response.data;

};

// ==========================================
// Set Password
// ==========================================

export const setPassword = async (data) => {

    const response = await api.post(
        "/property-admin/set-password",
        data
    );

    return response.data;

};